### ISSUE TITLE
Running "Clean Masters" on a plugin with 258 masters results in FormID errors

### ISSUE THREAD

**Issue by super-cooper** [TES5Edit/TES5Edit#932]
**What is the version of xEdit you are using?**
SSEEdit 4.0.3 [Auto Load] (1CDF3AC7)

**Describe the bug**
Any plugin with a full master list (258 entries), when having its masters cleaned, will modify a bunch of records' FormIDs, which results in numerous errors for any other records that references those that were modified. The plugin I tested this on had FormIDs modified on records that were defined by other mods, and merely modified by my plugin.

**To reproduce**
1. Right click on on a plugin that has a full master list
2. Select "Clean Masters"
3. Observe that a number of records not defined by the cleaned plugin have their FormIDs modified

**Expected behavior**
The plugin should simply have its master list cleaned of all non-required masters, and not have _any_ records modified.

**Screenshots**
![image](https://user-images.githubusercontent.com/14908880/122306843-2d67cd00-ced7-11eb-8c7b-0c2c0ef05fb5.png)

![image](https://user-images.githubusercontent.com/14908880/122306793-1b862a00-ced7-11eb-871e-50406c135729.png)

![image](https://user-images.githubusercontent.com/14908880/122306805-1f19b100-ced7-11eb-9c89-d59907b47a12.png)

**Additional context**
I encountered this because I basically have a custom bash patch for my large load order. Over several weeks, I have taken it into the CK to make cell/worldspace edits, and the CK attaches any mods you load before the active plugin as a master. These masters have built up over time, but I just figured I'd clean them eventually. Now that the list has filled up, I have tried to clean them, and this bug occurs. I am not sure why records/formIDs are being modified at all, but it appears to be modifying records that are defined in other plugins (but modified by this plugin) to have formIDs **as if** they were defined in this plugin. I've used "Clean Masters" many times before with no issues except for when the master list is full.

**Comment by ElminsterAU** [comment]
258 means the list is over full. I've no idea how you created that file in the first place. It's already totally broken.

The limit is 256 master entries (00..FF).

**Comment by ElminsterAU** [comment]
Eh, actually, the limit is 255 masters, as FF needs to remain free for records belonging to the file.

**Comment by super-cooper** [comment]
> 258 means the list is over full. I've no idea how you created that file in the first place. It's already totally broken.
> 
> The limit is 256 master entries (00..FF).

I've created the file entirely with xEdit and CK. Perhaps there is a bug in xEdit that allows too many masters to be added. I've begun manually looking for files that can be removed. Also, strangely enough, the game loads this plugin just fine. 

Anyway, would it be easy to manually cut the master list down in a hex editor? I am hoping to just get it down to 255 so that I can have xEdit do the rest. Or do you think I could manually remove them from the parents list in the CK and then save it there? I  am not sure if that removes masters or just chooses to not load them for the current editing session.

**Comment by ElminsterAU** [comment]
xEdit should under all circumstances (except for manual editing the master list in the file header) prevent more than 255 master from being added. I've no idea about CK.

Trying to fix that file now is the equivalent of open heart surgery without anaesthetics, good luck. I'm not going to touch that with a 10m pole.

**Comment by super-cooper** [comment]
I'm pretty sure the only issue is the master list. Very few of the files in the master list define records used by the plugin. I've never edited this file using anything except xEdit and the CK. I've also checked for errors, and it reports none. My question again is, is it easy to remove a master file that is not referenced anywhere else in the plugin by using a hex editor?

**Comment by ElminsterAU** [comment]
And I'm very sure that the master list affects every single form ID stored in the file, because the form ID xEdit or CK are showing you are not the ones stored in the file. The form ID stored in the file are mapped by looking up their high byte as index into the master list and replacing it with the current global load order form id prefix of that file.

As I said, I'm very sure no version of xEdit will add more than 255 masters, and I've no idea what CK does. But the file is fundamentally broken. 

This is my last comment in this topic.

**Comment by super-cooper** [comment]
> This is my last comment in this topic.

Fair enough.

Figured I would post here for anyone else who somehow finds themselves in the same situation:
I fixed this by going through each record type in my broken plugin (Activator, Ingestible, etc.) and deep-copy-as-override-ing them into a single new plugin. This way, only the necessary masters were added. xEdit reports zero errors, and the plugin loads fine in game and in the CK. 

Future advice to myself and others would be to clean your masters after doing edits in the CK on a full load order patch like this one.

Thank you @ElminsterAU for making xEdit so robust as to allow me to do this.
